### PR TITLE
chore: Add docs and tests for sending custom headers

### PR DESCRIPTION
**Pull request by jimmyjames**
This PR adds documentation and testing for custom header functionality in the OpenFGA JavaScript SDK.

#### What problem is being solved?

SDK is missing documentation and tests for sending custom HTTP headers.

#### How is it being solved?

Updating documentation and adding unit tests.

#### What changes are made to solve it?

- Added a new "Custom Headers" section to the README with examples for both default headers (via `baseOptions`) and per-request headers
- Created test suite (`tests/headers.test.ts`) with 19 test cases covering all header scenarios including:
  - Default headers from client configuration
  - Per-request headers and precedence behavior
  - Header merging and override behavior
  - Content-Type header protection behavior
  - Edge cases and special scenarios

In addition to unit testing, manual testing was also done:
- verified client headers being sent
- verified per-request headers being sent
- verified per-request headers override client headers

## References

- https://auth0team.atlassian.net/browse/DXAZD-1899
- https://auth0team.atlassian.net/browse/DXAZD-1941
- https://github.com/openfga/sdk-generator/issues/569

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

## Summary by CodeRabbit

- Documentation
  - Expanded guidance on Custom Headers: how to set default/global headers during client setup and override them per request, with examples. Clarifies header precedence and that Content-Type defaults to application/json unless overridden.

- Tests
  - Added comprehensive test coverage for header handling across client operations, validating default vs per-request merge and precedence, case-insensitive behavior, Content-Type protections, and various edge cases (empty/undefined options, special characters, large header sets).

### CODE REVIEW

File: tests/headers.test.ts [hunk 1]
```diff
@@ -0,0 +1,743 @@
...
+      await fgaClient.check({
+        user: "user:test",
+        relation: "reader",
+        object: "document:test"
+      });
+
+      expect(scope.isDone()).toBe(true);
+    });
+
+    it("allows Content-Type override via per-request headers", async () => {
```
Comment (jimmyjames) [inline tests/headers.test.ts:412]: I don't know if we actually want this behavior. But this test demonstrates what the actual behavior is; any change to that behavior will be intentional and reflected in the test changes as needed.

**Review by rhamzeh — APPROVED** [review]
